tests/test_radiation.py

```diff
@@ -0,0 +1,98 @@
+"""Test module for abiotic.radiation.py."""
+
+import pytest
+
+# from core.constants import CONSTANTS as C  # this doesn't exist yet
+from virtual_rainforest.models.abiotic.radiation import (
+    ALBEDO_SHORTWAVE,
+    ALBEDO_VIS,
+    BEER_REGRESSION,
+    BOLZMAN_CONSTANT,
+    CANOPY_EMISSIVITY,
+    CELCIUS_TO_KELVIN,
+    CLOUDY_TRANSMISSIVITY,
+    FLUX_TO_ENERGY,
+    SECOND_TO_DAY,
+    SOIL_EMISSIVITY,
+    TRANSMISSIVITY_COEFFICIENT,
+)
+
+
+def test_import_constants():
+    """Test that constants were imported correctly."""
+    assert CLOUDY_TRANSMISSIVITY == 0.25
+    assert TRANSMISSIVITY_COEFFICIENT == 0.50
+    assert FLUX_TO_ENERGY == 2.04
+    assert BOLZMAN_CONSTANT == 5.67e-8
+    assert SOIL_EMISSIVITY == 0.95
+    assert CANOPY_EMISSIVITY == 0.95
+    assert BEER_REGRESSION == 2.67e-5
+    assert ALBEDO_VIS == 0.03
+    assert ALBEDO_SHORTWAVE == 0.17
+    assert CELCIUS_TO_KELVIN == 273.15
+    assert SECOND_TO_DAY == 86400
+
+
+def test_calc_ppfd():
+    """Simple check for correct numbers, better test to be decided."""
+    from virtual_rainforest.models.abiotic import radiation
+
+    dummy = radiation.Radiation(100)
+    tau = dummy.calc_ppfd(29376000, 1.0)
+    assert tau == pytest.approx(0.752, 0.1)
+    assert dummy.ppfd == pytest.approx(43.713, 0.1)
+
```

Comment:
A couple of options here on structuring tests. The first is using `pytest.fixtures` - these are re-usable bits of code you can share between tests. So you could do simplify all the tests to do this:

```python
@pytest.fixture
def radiation_fixture():
    """Create a reusable radiation fixture"""
    from virtual_rainforest.models.abiotic.radiation import Radiation

    return Radiation(100)

def test_calc_ppfd(radiation_fixture):
    """Simple check for correct numbers, better test to be decided."""
    tau = radiation_fixture.calc_ppfd(29376000, 1.0)
    assert tau == pytest.approx(0.752, 0.1)
    assert radiation_fixture.ppfd == pytest.approx(43.713, 0.1)
```

The other thing is that you're using scalars as your inputs to methods that support arrays and it is worth testing that array inputs work as expected. You could simply replace the test values, but I would be tempted to use parameterisation:

```python
@pytest.mark.parameterise(
    argnames="input,exp_tau,exp_ppfd",
    argvalues=[
        (29376000, 0.752, 43.713),
        (
            np.array([29376000, 29376000]),
            np.array([0.752, 0.752]),
            np.array([43.713, 43.713]),
        ),
    ],
)
def test_calc_ppfd(radiation_fixture, input, exp_tau, exp_ppfd):
    """Simple check for correct numbers, better test to be decided."""
    tau = radiation_fixture.calc_ppfd(input, 1.0)
    assert tau == pytest.approx(exp_tau, 0.1)
    assert radiation_fixture.ppfd == pytest.approx(exp_ppfd, 0.1)
```

Reply:
Thanks for this suggestion, I included something like this for all the tests :-)